HIP/hip-415.md

```diff
@@ -0,0 +1,255 @@
+working-group: Danno Ferrin<@shemnon>, Richard Bair<@rbair23>, Steven Sheehy<steven.sheehy@hedera.com>, Mitchell 
+Martin<mitch@swirlds.com>
+type: Standards Track 
+category: Core 
+needs-council-approval: Yes 
+status: Draft
+created: 2022-04-03
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/434
+updated: 2022-04-13
+requires: HIP-435
+---
+
+## Abstract
+
+Specifies how to introduce and formalize the concepts of Blocks in Hedera Hashgraph so that it can be used as a foundation on which further interoperability with existing DLT networks and infrastructure can be built.
+
+## Motivation
+
+The concept of blocks is a vital part of the existing Ethereum infrastructure and, as such, the introduction of blocks 
+in Hedera can be considered a foundational step towards greater interoperability with EVM based tooling, explorers, exchanges and wallet providers.
+
+## Rationale
+
+Hedera must have a single consistent answer to what a block number is, the hash and which transactions are included in that block number. This is required for two reasons:
+
+1. The Smart Contracts Service to have the context of a `number` and the `hash` of the current block while running the EVM bytecode.
+2. Mirror Nodes to have the information in order to implement standard JSON RPC endpoints and be able to answer queries such as:
+    1. Getting current block number
+    2. Getting block by hash/number
+    3. Getting a list of transactions for a block by number/hash
+    4. Getting logs based on block number filtering
+
+**Design Goal #1 Minimize Changes**
+
+The Block concept should fit naturally into the existing processes, mechanisms and state updates. It must keep the same responsibilities between consensus, services and mirror nodes.
+
+**Design Goal #2 Lightweight**
+
+The Block concept must not add a lot of complexity and performance overhead to the processing of transactions. It must have minimal impact on the TPS of the network.
+
+Based on the described design goals above, the outlined specification defines that block properties are to be computed and populated at different points in time and by different components of the network, based on their responsibility.
+
+## Specification
+
+**Definitions**
+
+- `block` → `Record file` containing all `Record Stream Objects` for a given time frame. Block times are to be 
+  at least `hedera.recordStream.logPeriod` seconds. The genesis block and therefore number (`blockNumber=0`) is 
+  considered the stream start date with the first RCD file exported from `services` nodes.
+- `block number` → consecutive number of the `Record file` that is being incremented by `1` for every new `Record 
+  file`. For already existing networks, this value will be initially bootstrapped through `Mirror Nodes` and after that 
+  maintained by services nodes.
+- `block hash` → `32 byte` prefix (out of `48 bytes`) of the running hash of the last `Record Stream Object` from 
```

Comment:
I confirmed in the mirror node database that the first 32 bytes of the hash are still unique across all rows.